editor/editor_file_system.cpp

```diff
@@ -961,6 +961,14 @@ void EditorFileSystem::_scan_new_dir(EditorFileSystemDirectory *p_dir, Ref<DirAc
 
 		if (fi->uid != ResourceUID::INVALID_ID) {
 			if (ResourceUID::get_singleton()->has_id(fi->uid)) {
+				// Restrict UID dupe warning to first-scan since we know there are no file moves going on yet.
+				if (first_scan) {
+					// Warn if we detect files with duplicate UIDs.
+					String other_path = ResourceUID::get_singleton()->get_id_path(fi->uid);
```

Comment:
```suggestion
					const String &other_path = ResourceUID::get_singleton()->get_id_path(fi->uid);
```

Reply:
get_id_path returns a String by value, so I can go ahead and make it const but can't assign it to a reference.